consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -113,15 +148,172 @@ class KafkaEventConsumer(private val consumer: KafkaConsumer<String, ByteArray>,
+            try {
+                consumerRecordsMap.forEach { action(it.key.topic(), it.value.map { JSONUtils.readValue<Any>(it.value()) }) }
+            } catch (e: Exception) {
+                // TODO add dead letter queue
+            }
+        }
+    }
+
+    override fun read(topicConfig: Map<String, Any>, action: (String, List<Any>) -> Unit) {
+        val kafkaTopicConfig = KafkaTopicConfig.fromMap(topicConfig)
+        if (kafkaTopicConfig.topicPartitionsMap.isEmpty()) {
+            readSimple(action)
+        } else {
+            readFromPartition(kafkaTopicConfig, action)
+        }
+    }
+
+    fun toConsumerRecordsMap(topicPartitionsMap: Map<TopicPartition, Long>,
+                             records: ConsumerRecords<String, ByteArray>)
+            : Map<TopicPartition, List<ConsumerRecord<String, ByteArray>>> = topicPartitionsMap
+                .mapValues {
+                    records.records(it.key)
+                }
+                .filterValues { it.isNotEmpty() }
+
+    fun setSeek(topicPartitionsMap: Map<TopicPartition, Long>) {
+        if (isSeekSet) {
+            return
+        }
+        isSeekSet = true
+        consumer.poll(0) // dummy call see: https://stackoverflow.com/questions/41008610/kafkaconsumer-0-10-java-api-error-message-no-current-assignment-for-partition
+        topicPartitionsMap.forEach {
+            when (it.value) {
+                -1L -> consumer.seekToBeginning(listOf(it.key))
+                -2L -> consumer.seekToEnd(listOf(it.key))
+                else -> consumer.seek(it.key, it.value)
+            }
+        }
+    }
+}
+
+class KafkaManualCommitEventConsumer(private val config: KafkaSinkConfiguration,
+                                     private val log: Log): KafkaAutoCommitEventConsumer(config, log) {
+
+    private val topicPartitionOffsetMap = ConcurrentHashMap<TopicPartition, OffsetAndMetadata>()
+
+    override fun start() {
+        if (topics.isEmpty()) {
+            log.info("No topics specified Kafka Consumer will not started")
+            return
+        }
+        this.consumer.subscribe(topics, StreamsConsumerRebalanceListener(topicPartitionOffsetMap, consumer, config.autoOffsetReset, log))
+    }
+
+    private fun readSimple(action: (String, List<Any>) -> Unit): Map<TopicPartition, OffsetAndMetadata> {
+        val topicMap = mutableMapOf<TopicPartition, OffsetAndMetadata>()
+        val records = consumer.poll(0)
+        if (!records.isEmpty) {
+            this.topics.forEach { topic ->
+                val topicRecords = records.records(topic)
+                val lastRecord = topicRecords.last()
+                val offsetAndMetadata = OffsetAndMetadata(lastRecord.offset(), "")
+                val topicPartition = TopicPartition(lastRecord.topic(), lastRecord.partition())
+                topicMap[topicPartition] = offsetAndMetadata
+                topicPartitionOffsetMap[topicPartition] = offsetAndMetadata
```

Comment:
I would make this clearer that you're writing to a instance variable, e.g. by using a named method.